### PR TITLE
RBAC: update data source permission API reference

### PR DESCRIPTION
**Pull request by IevaVasiljeva**
**What is this feature?**

Update data source permission API reference.

**Why do we need this feature?**

We are removing the old data source permission endpoints, so need to update the documentation to use the new endpoints.

**Who is this feature for?**

Developers who use our APIs.

**Which issue(s) does this PR fix?**:

Related to https://github.com/grafana/grafana-enterprise/pull/5880

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/developers/http_api/datasource_permissions.md [hunk 1]
```diff
@@ -268,31 +271,62 @@ Status codes:
...
+**Example request:**
+
+```http
+POST /api/access-control/datasources/my_datasource/builtInRoles/Viewer
 Accept: application/json
 Content-Type: application/json
 Authorization: Bearer eyJrIjoiT0tTcG1pUlY2RnVKZTFVaDFsNFZXdE9ZWmNrMkZYbk
+
+{
+  "permission": "",
```
Comment (eleijonmarck) [inline docs/sources/developers/http_api/datasource_permissions.md:328]: should it be some response herre?

Comment (IevaVasiljeva) [reply docs/sources/developers/http_api/datasource_permissions.md:328]: This is the case for removing a permission - to remove it we pass in
```
{"permission": ""}
```
and get back
```
{"message": "Permission removed"}
```

File: docs/sources/developers/http_api/datasource_permissions.md [hunk 2]
```diff
@@ -27,84 +27,137 @@ title: Datasource Permissions HTTP API
...
+| ---------------------------- | ------------------------------------------------------------------------------------------ |
+| datasources.permissions:read | datasources:\*<br>datasources:uid:\*<br>datasources:uid:my_datasource (single data source) |
 
 ### Examples
 
 **Example request:**
 
 ```http
-POST /api/datasources/1/enable-permissions
+GET /api/datasources/my_datasource/permissions HTTP/1.1
```
Comment (gamab) [inline docs/sources/developers/http_api/datasource_permissions.md:58]:
```suggestion
GET /api/access-control/datasources/my_datasource/permissions HTTP/1.1
```

File: docs/sources/developers/http_api/datasource_permissions.md [hunk 3]
```diff
@@ -27,84 +27,137 @@ title: Datasource Permissions HTTP API
...
+## Add or revoke access to a data source for a user
 
-## Disable permissions for a data source
+`POST /api/access-control/datasources/:uid/users/:id`
 
-`POST /api/datasources/:id/disable-permissions`
+Sets user permission for the data source with the given `uid`.
 
-Disables permissions for the data source with the given `id`. All existing permissions will be removed and anyone will be able to query the data source.
+To add a permission, set the `permission` field to either `Query` or `Edit`.
```
Comment (gamab) [inline docs/sources/developers/http_api/datasource_permissions.md:137]: Didn't you add `Admin` lately?

**Review by gamab — APPROVED** [review]
Great job! I think you might have missed one URL change. Also haven't you added an `Admin` access level lately?

File: docs/sources/developers/http_api/datasource_permissions.md (hunk 3 above)
Comment (IevaVasiljeva) [reply docs/sources/developers/http_api/datasource_permissions.md:132]: Well spotted! I had this PR open beforehands and forgot to update it.

File: docs/sources/developers/http_api/datasource_permissions.md [hunk 4]
```diff
@@ -27,84 +27,132 @@ title: Datasource Permissions HTTP API
...
-## Disable permissions for a data source
+## Add or revoke access to a data source for a user
 
-`POST /api/datasources/:id/disable-permissions`
+`POST /api/access-control/datasources/:uid/users/:id`
 
-Disables permissions for the data source with the given `id`. All existing permissions will be removed and anyone will be able to query the data source.
+Sets user permission for the data source with the given `uid`.
+
+To add a permission, set the `permission` field to either `Query` or `Edit`.
```
Comment (IevaVasiljeva) [inline docs/sources/developers/http_api/datasource_permissions.md:132]:
```suggestion
To add a permission, set the `permission` field to either `Query`, `Edit` or `Admin`.
```

File: docs/sources/developers/http_api/datasource_permissions.md [hunk 5]
```diff
@@ -114,115 +162,68 @@ HTTP/1.1 200 OK
...
-## Add permission for a data source
+## Add or revoke access to a data source for a team
 
-`POST /api/datasources/:id/permissions`
+`POST /api/access-control/datasources/:uid/teams/:id`
 
-Adds a user permission for the data source with the given `id`.
+Sets team permission for the data source with the given `uid`.
+
+To add a permission, set the `permission` field to either `Query` or `Edit`.
```
Comment (IevaVasiljeva) [inline docs/sources/developers/http_api/datasource_permissions.md:204]:
```suggestion
To add a permission, set the `permission` field to either `Query`, `Edit`, or `Admin`.
```

File: docs/sources/developers/http_api/datasource_permissions.md [hunk 6]
```diff
@@ -268,31 +266,39 @@ Status codes:
...
 
-`DELETE /api/datasources/:id/permissions/:permissionId`
+`POST /api/access-control/datasources/:uid/builtInRoles/:builtinRoleName`
 
-Removes the permission with the given `permissionId` for the data source with the given `id`.
+Sets permission for the data source with the given `uid` to all users who have the specified basic role.
+
+You can set permissions for the following basic roles: `Admin`, `Editor`, `Viewer`.
+
+To add a permission, set the `permission` field to either `Query` or `Edit`.
```
Comment (IevaVasiljeva) [inline docs/sources/developers/http_api/datasource_permissions.md:278]:
```suggestion
To add a permission, set the `permission` field to either `Query`, `Edit`, or `Admin`.
```

File: docs/sources/developers/http_api/datasource_permissions.md [hunk 7]
```diff
@@ -268,31 +266,39 @@ Status codes:
...
 
 ```http
-DELETE /api/datasources/1/permissions/2
+POST /api/access-control/datasources/my_datasource/builtInRoles/Admin
 Accept: application/json
 Content-Type: application/json
 Authorization: Bearer eyJrIjoiT0tTcG1pUlY2RnVKZTFVaDFsNFZXdE9ZWmNrMkZYbk
+
+{
+  "permission": "Edit",
```
Comment (IevaVasiljeva) [inline docs/sources/developers/http_api/datasource_permissions.md:300]:
```suggestion
  "permission": "Admin",
```

**Pull request merged**